File: data/multilang_comparison/program_cpp_vector/src/func37.cpp
#include "program_cpp_vector.hpp" 
Array* func37(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   unsigned int loop17 = 0;
   unsigned int loopLimit17 = (50)/4 + 1;
   for(; loop17 < loopLimit17; loop17++) {
   }
   Array* array50;
   if (pCounter > 0) {
      array50 = vars->data.at(--pCounter);
      array50->refC++;
      DEBUG_COPY(array50->id);
   } else {
      array50 = new Array();
      array50->size = 859;
      array50->refC = 1;
      array50->id = 50;
      array50->data = std::vector<unsigned int>(array50->size, 0);
      DEBUG_NEW(array50->id);
   }
   return array50;
}

File: data/multilang_comparison/program_go/src/func37.go
package main
import "math/rand"
var _ = rand.Intn
func Func37(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var loop17 int = 0;
   var loopLimit17 int = (50)/4 + 1;
   for ; loop17 < loopLimit17; loop17++ {
   }
   var array50 *Array
   if pCounter > 0 {
       array50 = vars.Data[pCounter-1]
       pCounter--
       array50.RefC++
   } else {
       array50 = &Array{
           Size: 859,
           RefC: 1,
           Id: 50,
           Data: make([]uint32, 859),
       }
   }
   return array50;
}

